src/core/shaders/index.js

```diff
@@ -34,6 +36,7 @@ function compileProgram(gl, glslVS, glslFS) {
     if (!gl.getProgramParameter(this.program, gl.LINK_STATUS)) {
         throw new Error('Unable to link the shader program: ' + gl.getProgramInfoLog(this.program));
     }
+    this.programID = programID++;
```

Comment:
This is not used

Reply:
"Improve shader program debugging", WebGL shaders don't have even an ID, it is a nightmare for debugging purposed if we don't add some IDs.

Reply:
what about a comment?

`// programID is used to improve shader debugging`